### DISCUSSION TITLE
types/react-redux static defaultProps are not working anymore with version 7.1.18

### DISCUSSION THREAD

**Discussion by himanshumeena** [DefinitelyTyped/DefinitelyTyped#54535] [Issues with a @types package]
hi 
after upgrading types/react-redux from version 7.1.17 to 7.1.18 my static defaultProps arenn't working anymore.

example:
i am having this component usage
```
<Carousel
      componentId={componentId}
      instrumentId={
          carousel.instrumentId
      }
      instrument={instrumentData as any}
      resizeTimestamp={resizeTimestamp}
      height={carouselHeight}
      watchlistId={watchlistId}
      onTriggerMenu={
          this.handleTriggerMenu
      }
  />
  ```
  Carousel component also have 2 more props which are functions `onAnimationBegin` and `onAnimationBegin` which are defined in static defaultProps like this:
  ```
  static defaultProps = {
        step: 0,
        height: 0,
        isTradeSignalsAllowed: false,
        onStepChange: _.noop,
        onAnimationBegin: _.noop,
        onAnimationComplete: _.noop,
        onTriggerAction: _.noop,
        onTriggerMenu: _.noop,
    };
  ```
  and ownProps declaration for Carousel component looks like this
  ```
  export type OwnProps = Readonly<{
    componentId: string;
    instrumentId: string;
    instrument: instrumentTypes.Instrument;
    resizeTimestamp: number;
    height: number;
    watchlistId: string;
    onAnimationBegin: () => void;
    onAnimationComplete: () => void;
    onTriggerMenu: (arg0: Event) => void;
}>;
  ```

the error i am getting on the console is this after running ts
```
 error TS2739: Type '{ componentId: string; instrumentId: string; instrument: any; resizeTimestamp: number; height: number; watchlistId: string; onTriggerMenu: () => void; }' is missing the following properties from type 'Readonly<{ componentId: string; instrumentId: string; instrument: Instrument; resizeTimestamp: number; height: number; watchlistId: string; onAnimationBegin: () => void; onAnimationComplete: () => void; onTriggerMenu: (arg0: Event) => void; }>': onAnimationBegin, onAnimationComplete

863                                                     <Carousel
```
  
so i was hoping it should have picked the missing props `onAnimationBegin` and `onAnimationComplete` from the static defaultProps but it didn't 

i don't want to pass these missing props since it defeats the static defaultPorps usecase then

let me know if i am missing something here or something changed with the new version 
would help if you can give some example to the new way of using defaultProps if any

**Comment by lukeapage** [comment]
This is not a bug - after the update you now have to explicitly give a container the correct types e.g. if there is a default prop on the container, the ownprop type must be `{ prop?: string }` and then the components prop can be `{ prop: string }` as there is a default prop on the component. This is more correct - the default prop doesn't get applied inside the container.
